File: Programing_Basics_with_JS/whileLoopMoreExercises/01.Dishwasher.js
function dishwasher(input) {
    let index = 0;
    let detergentBottles = Number(input[index]);
    index++;

    let volumeOfBottle = 750;
    let potNeededDetergent = 15;
    let plateNeededDetergent = 5;
    let totalDetergent = detergentBottles * volumeOfBottle;
    let dishwasherloadingCount = 0;
    let command = input[index];
    index++;
    let plateCount = 0;
    let potCount = 0;


    while(command !== 'End') {
        let numberOfVessels = Number(command);
        dishwasherloadingCount++;

        if(dishwasherloadingCount % 3 === 0) {
            potCount += numberOfVessels;
            totalDetergent -= numberOfVessels * potNeededDetergent;
        } else {
            plateCount += numberOfVessels;
            totalDetergent -= numberOfVessels * plateNeededDetergent;
        }

        if(totalDetergent < 0) {
            break;
        }

        command = input[index];
        index++;
    }

    if(totalDetergent >= 0) {
        console.log('Detergent was enough!');
        console.log(`${plateCount} dishes and ${potCount} pots were washed.`);
        console.log(`Leftover detergent ${totalDetergent} ml.`);
    } else {
        console.log(`Not enough detergent, ${Math.abs(totalDetergent)} ml. more necessary!`);
    }
}

dishwasher(['2', '53', '65', '55', 'End'])
dishwasher(['1', '10', '15', '10', '12', '13', '30'])

File: Programming_Basics_with_Python/while_loop_more_exercises/01._dishwasher.py
bottle_with_detergent = int(input())
command = input()

bottle_value = 750
soap_for_one_plate = 5
soap_for_one_pot = 15
machine_count = 0
clean_pots_count = 0
clean_plates_count = 0

all_soap = bottle_with_detergent * bottle_value

while not command == 'End':
    dishes_for_wash = int(command)
    machine_count += 1

    if machine_count % 3 == 0:
        all_soap -= dishes_for_wash* soap_for_one_pot
        clean_pots_count += dishes_for_wash
    else:
        all_soap -= dishes_for_wash * soap_for_one_plate
        clean_plates_count += dishes_for_wash

    if all_soap < 0:
        break

    command = input()

if all_soap >= 0:
    print('Detergent was enough!')
    print(f'{clean_plates_count} dishes and {clean_pots_count} pots were washed.')
    print(f'Leftover detergent {all_soap} ml.')
else:
    print(f'Not enough detergent, {abs(all_soap)} ml. more necessary!')
